src/network/networkd-dhcp4.c

```diff
@@ -1660,6 +1705,47 @@ int dhcp4_update_mac(Link *link) {
         return 0;
 }
 
+static int dhcp4_ipv4_only_preferred_expire_handler(sd_event_source *s, uint64_t usec, void *userdata) {
+        Link *link = ASSERT_PTR(userdata);
+        usec_t now_usec;
+
+        assert(link->manager);
+
+        assert_se(sd_event_now(link->manager->event, CLOCK_BOOTTIME, &now_usec) >= 0);
+
+        (void) dhcp4_start(link);
+
+        return 0;
+}
+
+int dhcp4_start_delayed(Link *link, uint32_t delay) {
```

Comment:
We have `client_start_delayed()` in `sd-dhcp-client.c`. I think this is not necessary.

Reply:
There were few things here.
1. We want to handle it outside of the library there is a follow up part 
```
If the client is in the INIT-REBOOT state, it SHOULD stop the DHCPv4 configuration process or disable the IPv4 stack completely for V6ONLY_WAIT seconds or until the network attachment event, whichever happens first
```
Here it stopped completely. 

```
until the network attachment event, whichever happens first
```
I am not  clear about this . it talks about https://datatracker.ietf.org/doc/html/rfc4957. 

I left this TBD later. we later can work on this . `client_start_delayed` has few challenges too.

Reply:
I mean, this should be handled in sd-dhcp-client.c, rather than networkd-dhcp4.c.

The `sd_dhcp_client` object already has several timer event sources. So, please introduce another timer event source to suspend the received timespan.

And, in networkd side, if `AllowIPv6OnlyMode=` is enabled, simply call `sd_dhcp_client_stop()` when an IPv6 network connectivity is configured.